README.md

```diff
@@ -1,3 +1,77 @@
 # terraform-aws-ldap-maintainer
 
 A step function to maintain LDAP users via slack.
+
+## Overview
+
+This project deploys a collection of lambda functions, an api gateway endpoint, and a step function implemented with the [callback pattern](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-wait-token) that will automate disabling LDAP users via an interactive slack message.
+
+## Architecture
+
+![State Machine Definition](_docs/state_machine_def_0.0.1.png)
+
+## Submodules
+
+- [API Gateway](/modules/api_gateway): An API endpoint that responds asynchronously to slack events and triggers the Slack Listener lambda function
+- [LDAP Query](/modules/lambda_functions/ldap_query): Lambda function used to perform actions against a target ldap database
+- [Slack Listener](/modules/lambda_functions/slack_listener): Lambda function that responds to slack events via an asynchronously executed lambda function
```

Comment:
If each of these lambda function are in fact standalone modules then I think they should be a level higher right? i.e., /modules/api_gateway, /modules/ldap_query, etc)

Reply:
I suppose? It just made it easier to differentiate which modules were lambda related and which ones weren't.

Reply:
moved the modules to a common dir